sys/net/application_layer/cord/lc/cord_lc.c

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (C) 2017-2018 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_cord_lc
+ * @{
+ *
+ * @file
+ * @brief       CoRE Resource Directory lookup implementation
+ *
+ * @author      Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "mutex.h"
+#include "assert.h"
+#include "thread_flags.h"
+
+#include "net/gcoap.h"
+#include "net/cord/lc.h"
+
+#define FLAG_SUCCESS        (0x0001)
+#define FLAG_TIMEOUT        (0x0002)
+#define FLAG_ERR            (0x0004)
+#define FLAG_OVERFLOW       (0x0008)
+#define FLAG_NORSC          (0x0010)
+#define FLAG_MASK           (0x00ff)
+
+#define BUFSIZE             (512U)
+
+static char *_res_buf;
+static size_t _res_buf_len;
+
+static char _rd_lookif[NANOCOAP_URI_MAX];
+
+static mutex_t _mutex = MUTEX_INIT;
+static volatile thread_t *_waiter;
+
+static uint8_t buf[BUFSIZE];
+
+static void _lock(void)
+{
+    mutex_lock(&_mutex);
+    _waiter = sched_active_thread;
+}
+
+static int _sync(void)
```

Comment:
How about moving both `_lock` and `_sync` to `cord_common` and rename to `cord_common_lock` and `cord_common_sync`?

Reply:
It would still be nice not to duplicate this code, but then again I don't see this as a show stopper for now. So lets postpone this step until after this PR is merged...